main.js

```diff
@@ -2,52 +2,53 @@ var apply = Function.prototype.apply;
 
 // DOM APIs, for completeness
 
-exports.setTimeout = function() {
-  return new Timeout(apply.call(setTimeout, window, arguments), clearTimeout);
+exports.setTimeout = function () {
+	return new Timeout(apply.call(setTimeout, window, arguments), clearTimeout);
 };
-exports.setInterval = function() {
-  return new Timeout(apply.call(setInterval, window, arguments), clearInterval);
+exports.setInterval = function () {
+	return new Timeout(apply.call(setInterval, window, arguments), clearInterval);
 };
 exports.clearTimeout =
-exports.clearInterval = function(timeout) {
-  if (timeout) {
-    timeout.close();
-  }
-};
+	exports.clearInterval = function (timeout) {
+		if (timeout) {
+			timeout.close();
+		}
+	};
 
 function Timeout(id, clearFn) {
-  this._id = id;
-  this._clearFn = clearFn;
+	this._id = id;
+	this._clearFn = clearFn;
 }
-Timeout.prototype.unref = Timeout.prototype.ref = function() {};
-Timeout.prototype.close = function() {
-  this._clearFn.call(window, this._id);
+Timeout.prototype.unref = Timeout.prototype.ref = function () { };
+Timeout.prototype.close = function () {
+	this._clearFn.call(window, this._id);
 };
 
 // Does not start the time, just sets up the members needed.
-exports.enroll = function(item, msecs) {
-  clearTimeout(item._idleTimeoutId);
-  item._idleTimeout = msecs;
+exports.enroll = function (item, msecs) {
+	clearTimeout(item._idleTimeoutId);
+	item._idleTimeout = msecs;
 };
 
-exports.unenroll = function(item) {
-  clearTimeout(item._idleTimeoutId);
-  item._idleTimeout = -1;
+exports.unenroll = function (item) {
+	clearTimeout(item._idleTimeoutId);
+	item._idleTimeout = -1;
 };
 
-exports._unrefActive = exports.active = function(item) {
-  clearTimeout(item._idleTimeoutId);
+exports._unrefActive = exports.active = function (item) {
+	clearTimeout(item._idleTimeoutId);
 
-  var msecs = item._idleTimeout;
-  if (msecs >= 0) {
-    item._idleTimeoutId = setTimeout(function onTimeout() {
-      if (item._onTimeout)
-        item._onTimeout();
-    }, msecs);
-  }
+	var msecs = item._idleTimeout;
+	if (msecs >= 0) {
+		item._idleTimeoutId = setTimeout(function onTimeout() {
+			if (item._onTimeout)
+				item._onTimeout();
+		}, msecs);
+	}
 };
 
 // setimmediate attaches itself to the global object
 require("setimmediate");
-exports.setImmediate = setImmediate;
-exports.clearImmediate = clearImmediate;
+var global = require('global');
+exports.setImmediate = global.setImmediate || global.setTimeout;
```

Comment:
Why do you need `setTimeout` here? Is `setImmediate` not actually attaching to the global at all?

If so, it sounds like something needs to be fixed in the `setimmediate` library first. It doesn't seem to return anything on require, so it seems like it needs to be convinced to attach.